File: scripts/flutter/template/flutter_route_template.py
#!/usr/bin/env python3
"""Flutter GoRoute 템플릿 생성 스크립트"""

import argparse


def generate_route(
    screen_name: str,
    path: str = "/",
    no_transition: bool = False,
) -> str:
    """GoRoute 템플릿 코드 생성

    Args:
        screen_name: 스크린 이름 (PascalCase, 예: Login, Home)
        path: 라우트 경로 (예: /, /home)
        no_transition: True이면 pageBuilder + NoTransitionPage 래핑
    """
    if no_transition:
        return f"""GoRoute(
        path: '{path}',
        pageBuilder: (BuildContext context, GoRouterState state) {{
          return NoTransitionPage<void>(
            child: MultiBlocProvider(
              providers: [
                BlocProvider(
                  create: (_) => sl<{screen_name}Bloc>()..add({screen_name}LoadRequested()),
                ),
              ],
              child: const {screen_name}Screen(),
            ),
          );
        }},
      ),"""

    return f"""GoRoute(
        path: '{path}',
        builder: (BuildContext context, GoRouterState state) {{
          return MultiBlocProvider(
            providers: [
              BlocProvider(
                create: (_) => sl<{screen_name}Bloc>()..add({screen_name}LoadRequested()),
              ),
            ],
            child: {screen_name}Screen(),
          );
        }},
      ),"""


def main():
    parser = argparse.ArgumentParser(description="Flutter GoRoute 템플릿 생성")
    parser.add_argument("screen_name", help="스크린 이름 (PascalCase, 예: Login)")
    parser.add_argument("path", nargs="?", default="/", help="라우트 경로 (기본: /)")
    parser.add_argument(
        "-nt",
        action="store_true",
        help="NoTransitionPage 래핑 (pageBuilder 사용)",
    )
    args = parser.parse_args()

    print(generate_route(args.screen_name, args.path, args.nt))


if __name__ == "__main__":
    main()

File: scripts/flutter/template/flutter-route-template.mjs
#!/usr/bin/env node
// Flutter GoRoute 템플릿 생성 스크립트

import process from 'node:process';

function generateRoute(screenName, routePath = '/', noTransition = false) {
  if (noTransition) {
    return `GoRoute(
        path: '${routePath}',
        pageBuilder: (BuildContext context, GoRouterState state) {
          return NoTransitionPage<void>(
            child: MultiBlocProvider(
              providers: [
                BlocProvider(
                  create: (_) => sl<${screenName}Bloc>()..add(${screenName}LoadRequested()),
                ),
              ],
              child: const ${screenName}Screen(),
            ),
          );
        },
      ),`;
  }

  return `GoRoute(
        path: '${routePath}',
        builder: (BuildContext context, GoRouterState state) {
          return MultiBlocProvider(
            providers: [
              BlocProvider(
                create: (_) => sl<${screenName}Bloc>()..add(${screenName}LoadRequested()),
              ),
            ],
            child: ${screenName}Screen(),
          );
        },
      ),`;
}

function printUsage() {
  process.stderr.write('usage: flutter-route-template.mjs [-h] [-nt] screen_name [path]\n');
  process.stderr.write('\nFlutter GoRoute 템플릿 생성\n');
  process.stderr.write('\npositional arguments:\n');
  process.stderr.write('  screen_name  스크린 이름 (PascalCase, 예: Login)\n');
  process.stderr.write('  path         라우트 경로 (기본: /)\n');
  process.stderr.write('\noptions:\n');
  process.stderr.write('  -h, --help   show this help message and exit\n');
  process.stderr.write('  -nt          NoTransitionPage 래핑 (pageBuilder 사용)\n');
}

function main() {
  const argv = process.argv.slice(2);
  let noTransition = false;
  const positional = [];

  for (const arg of argv) {
    if (arg === '-h' || arg === '--help') {
      printUsage();
      process.exit(0);
    } else if (arg === '-nt') {
      noTransition = true;
    } else if (arg.startsWith('-')) {
      process.stderr.write(`error: unrecognized arguments: ${arg}\n`);
      process.exit(2);
    } else {
      positional.push(arg);
    }
  }

  if (positional.length < 1) {
    process.stderr.write('error: the following arguments are required: screen_name\n');
    process.exit(2);
  }
  if (positional.length > 2) {
    process.stderr.write(`error: unrecognized arguments: ${positional.slice(2).join(' ')}\n`);
    process.exit(2);
  }

  const screenName = positional[0];
  const routePath = positional[1] ?? '/';
  console.log(generateRoute(screenName, routePath, noTransition));
}

main();
